Return to the correct sub step in VBA flow after error in adding Bank Account manually

aldo-expensify opened the pull request:
cc @marcaaron: copying Mark here because he may know the reason why we are copying this value to a local state.

### Details

The subStep (`reimbursementAccount.achData.subStep`) in the VBA flow was being copied from Onyx into a local state in the `BankAccountStep` component ([here](https://github.com/Expensify/App/blob/82e1b44901c32d699a8cd3facaeb593d4d9c34ef/src/pages/ReimbursementAccount/BankAccountStep.js#L54)). Then, when rendering `BankAccountStep`, we read the `subStep` using `this.state.bankAccountAddMethod ` (i.e. [here](https://github.com/Expensify/App/blob/3f5f90d6f2dc1653cffe35e42c4a907f0ac6e33a/src/pages/ReimbursementAccount/BankAccountStep.js#L249)).

The problem with this approach is that we have now to sources of truth for the `subStep`:
1. Onyx (`reimbursementAccount.achData.subStep`)
2. [Local state](https://github.com/Expensify/App/blob/82e1b44901c32d699a8cd3facaeb593d4d9c34ef/src/pages/ReimbursementAccount/BankAccountStep.js#L54) in the component

This can potentially cause bugs when these states are not synchronized properly.

I this case:

1. When we click "Connect Manually" in the `BankAccountStep`, we just update the `subStep` in the local state ([here](https://github.com/Expensify/App/blob/3f5f90d6f2dc1653cffe35e42c4a907f0ac6e33a/src/pages/ReimbursementAccount/BankAccountStep.js#L226)), Onyx doesn't know about this update. 
2. When we submit the form with the "Routing Number" and "Account Number", we call [this method](https://github.com/Expensify/App/blob/3f5f90d6f2dc1653cffe35e42c4a907f0ac6e33a/src/pages/ReimbursementAccount/BankAccountStep.js#L139-L155) which will end up setting `reimbursementAccount.loading` to `true` in Onyx ([here](https://github.com/Expensify/App/blob/6cdfd4b9003e74d4ea1806f79ce16485f778ae6e/src/libs/actions/BankAccounts.js#L615))
3. `reimbursementAccount.loading` with `true` value in Onyx will make the loading indicator appear ([here](https://github.com/Expensify/App/blob/6cdfd4b9003e74d4ea1806f79ce16485f778ae6e/src/pages/ReimbursementAccount/ReimbursementAccountPage.js#L176-L178)), destroying the `BankAccountStep` and its local state.
4. Once the loading finishes and the backend returns an error, `reimbursementAccount.loading` is set to `false` [here](https://github.com/Expensify/App/blob/6cdfd4b9003e74d4ea1806f79ce16485f778ae6e/src/libs/actions/BankAccounts.js#L652-L655)
5. `ReimbursementAccountPage` removes `VBALoadingIndicator` and creates `BankAccountStep` step again.
6. `BankAccountStep` copies again the `subStep` to its local state (constructor called again). The `subStep` is `null` at this point.
7. The user is taken back to the screen where he chooses "Connect Manually" or plaid.

There can be more than one approach to solve this, but here I'll mention two:
1. Get rid of the multiple sources of truth so we don't have to care about any possible desynchronization 
2. Make sure that Onyx gets updated properly when we update the local state

The implementation I'm trying to do is (1.) because in my experience dealing with states that get copied and have to be updated together makes things more complex, and I prefer to avoid it if possible. It can cause bugs like this that are not so easy to catch and it is hard to see all cases where things could become unsynchronized .

When clearing the `subStep` from Onyx, I was only able to do so setting it to `null`. Setting it to `undefined` didn't do anything.


### Fixed Issues

$ https://github.com/Expensify/App/issues/5028

## Tests AND QA

### Case 1: Invalid routing number error

1. Navigate to /bank-account/
2. Click "Connect manually"
3. Input an invalid routing number, i.e. `12312312344`
4. Input any bank number, i.e. `123`
5. Click "Save & continue" button

Expected results:

- We should see the loading indicator (form not visible)
- The form showing the "Routing Number" and "Bank number" should appear again with an inline error for the "Routing number".

![Screen Shot 2021-09-22 at 11 00 05 AM](https://user-images.githubusercontent.com/87341702/134397030-c0cd258c-1487-4ecc-a5e9-b4f85f1e13a3.png)

### Case 2: Unverified account error

1. Create an account using OldDot (enter email, click Join)
2. Choose "Set up my company for free" (seems like you only get these options on web mobile)
![Screen Shot 2021-09-23 at 2 45 06 PM](https://user-images.githubusercontent.com/87341702/134588357-bfdf862f-6f4c-4e87-915b-e86cac9bd44d.png)
3. You will be taken to NewDot
4. Click "Expensify Card", then "Get Started"
5. In the "Add Bank Account" step (first), click "Connect manually"
6. Input any routing number / bank number
7. Click "Save & Continue"

Expected results
- We should see the loading indicator (form not visible)
- The form appears again with an error at the bottom about the account being unverified.

![Screen Shot 2021-09-23 at 2 50 34 PM](https://user-images.githubusercontent.com/87341702/134588931-702d3ff2-009a-4ca6-9083-e98167cdd04f.png)




### Tested On

- [x] Web
- [ ] Mobile Web


- [ ] Desktop
- [ ] iOS
- [x] Android

### Screenshots
<!-- Add screenshots for all platforms tested. Pull requests won't be merged unless the screenshots show the app was tested on all platforms.-->

#### Web
<!-- Insert screenshots of your changes on the web platform-->

#### Mobile Web
<!-- Insert screenshots of your changes on the web platform (from a mobile browser)-->

#### Desktop
<!-- Insert screenshots of your changes on the desktop platform-->

#### iOS
<!-- Insert screenshots of your changes on the iOS platform-->

#### Android
<!-- Insert screenshots of your changes on the Android platform-->

aldo-expensify committed:
Handle the subStep directly from Onyx

The subState in the VBA flow was being copied from Onyx into a local
state in the BankAccountStep component and it was read from this local
state. This caused that when we update the subState from BankAccounts.js
in Onyx, the component BankAccountStep missed the update because
it was reading it from the local state.

aldo-expensify committed:
Avoid double-fetching the withdrawal account data

Withdrawal account data was being loaded from AuthScreens component,
and then again from ReiumbursementAccountPage on mount. Moved the
code handling the step from route to componentDidUpdate in
ReimbursementAccountPage and is not fetching anymore.

aldo-expensify committed:
Merge branch 'main' of github.com:Expensify/App into aldo_fix-wrong-substep-after-error

aldo-expensify commented:
Hey @marcaaron , can you have a look at this? I think this fixes the problem and avoid some unnecessary double fetching of data from the backend, but then maybe I'm missing something.

marcaaron commented:
Hey @aldo-expensify, I can take a look at this but I'm having some trouble following the explanation in the details section. Can you try using a bulleted list with links or code snippets ([permalinks](https://docs.github.com/en/github/writing-on-github/working-with-advanced-formatting/creating-a-permanent-link-to-a-code-snippet)) to specific things in the code you are calling out?

marcaaron commented on src/libs/actions/BankAccounts.js:
```diff
@@ -339,9 +339,9 @@ function fetchFreePlanVerifiedBankAccount(stepToOpen) {
+        loading: true, // shows loading indicator
         error: '',
-
+        fetching: true, // only used when fetching
```
Why do we need both `fetching` and `loading`? The comments here don't really explain the difference or when we'd want to refer to one or the other or why.

aldo-expensify replied:
This doesn't feel that clean to me either, but the purpose is this:

I think we were doing the second fetch just with the purpose of reacting to the route parameter and show the right screen when we open directly a route like `/bank-account/company`

Since I'm avoiding to fetch the information a second time in the `ReimbursementAccountPage`'s constructor, I needed a way to know that the first fetch had finished to trigger the check about the route parameter ([here](https://github.com/Expensify/App/blob/251ee9719c37a3bd0259c89a31914bec0c964295/src/pages/ReimbursementAccount/ReimbursementAccountPage.js#L106))

I didn't use the `loading` state because that is used to [show the loading indicator](https://github.com/Expensify/App/blob/main/src/pages/ReimbursementAccount/ReimbursementAccountPage.js#L176-L178), and it is not used only when fetching, we also set it to `loading: true` when submitting information to the server. I don't want to trigger the [code related to the route parameter](https://github.com/Expensify/App/blob/251ee9719c37a3bd0259c89a31914bec0c964295/src/pages/ReimbursementAccount/ReimbursementAccountPage.js#L106)) when we submit information.

Is my assumption correct about this code only being there to handle the application starting from one of these routes? If I remove that code, everything seems to work fine, except for if I open the application from `/bank-account/company`, I get the screen to choose between manual and plaid.

aldo-expensify replied:
Removed this, if we decide to work in removing the second fetch, it will be in another PR.

marcaaron commented on src/pages/ReimbursementAccount/BankAccountStep.js:
```diff
@@ -80,6 +79,16 @@ class BankAccountStep extends React.Component {
     }
 
+    setSubStep(subStep) {
+        console.debug('setSubStep', subStep);
```
If we are planning to leave this in we might want to give it a better text description like 

`[BankAccountStep] sub step selected: ${subStep}`

marcaaron commented on src/pages/ReimbursementAccount/BankAccountStep.js:
```diff
@@ -80,6 +79,16 @@ class BankAccountStep extends React.Component {
+            // undefined seems to be ignored by Onyx
+            Onyx.merge(ONYXKEYS.REIMBURSEMENT_ACCOUNT, {achData: {subStep: null}});
+        } else {
+            Onyx.merge(ONYXKEYS.REIMBURSEMENT_ACCOUNT, {achData: {subStep}});
```
As a convention we avoid using `Onyx.merge()` outside of an actions file. I would recommend checking out the readme before continuing further. https://github.com/Expensify/App#structure-of-the-app

aldo-expensify replied:
You are right. I read that documentation, but I forgot about that. I'll have it more present now, thanks!

marcaaron commented on src/pages/ReimbursementAccount/BankAccountStep.js:
```diff
@@ -191,15 +200,16 @@ class BankAccountStep extends React.Component {
         // Disable bank account fields once they've been added in db so they can't be changed
         const isFromPlaid = this.props.achData.setupType === CONST.BANK_ACCOUNT.SETUP_TYPE.PLAID;
         const shouldDisableInputs = Boolean(this.props.achData.bankAccountID) || isFromPlaid;
+        const {subStep} = this.props.achData;
```
Some guidelines on destructuring here - https://github.com/Expensify/App/blob/main/STYLE.md#destructuring

Probably would help to read the entire style guide while you are at it if you haven't yet :)

aldo-expensify replied:
I did! is just that it takes time to remember all rules from a guideline, with time you should have to correct less and less of this type of errors 😬

marcaaron reviewed (changes requested):
I took a quick look at this PR. I haven't yet taken the time to understand what it is doing or why but just ran through to give some suggestions on style and convention. Thanks!

aldo-expensify commented:
> Hey @aldo-expensify, I can take a look at this but I'm having some trouble following the explanation in the details section. Can you try using a bulleted list with links or code snippets ([permalinks](https://docs.github.com/en/github/writing-on-github/working-with-advanced-formatting/creating-a-permanent-link-to-a-code-snippet)) to specific things in the code you are calling out?

ok, I'll do that!

aldo-expensify commented:
It was true that the details were a bit cryptic, I had to debug it again to understand what was going on. Hopefully the new explanation is easier to understand.

marcaaron commented:
>Get rid of the multiple sources of truth so we don't have to care about any possible desynchronization

Ok I read the first half of what you've wrote and I think I agree that it would be best to only rely on the `achData` Onyx to tell us which `subStep` we should show.

>On a side note, I see that if we get to the Company Step, then we close the sidebar, then click again in the Workspace's "Get Started" button, we still start from the beginning. This made me doubt about the purpose of keeping the subStep and currentStep (or any other "step" like information) in Onyx. 

We need to know the `currentStep` since bank accounts are saved in the server with information about which step the user needs to finish completing. If you check out the API layer this part will make more sense. As for the `subStep` I don't _think_  the API cares about it (but not sure) and it's mostly a way to tell the front end to show the plaid or manual flows.

>Are we keeping it locally because we destroy step components when we show the VBALoadingIndicator (here)?

I don't really know for sure to tell you the truth. If I had to guess we did it this way because it's not easy to set just one property of the `achData` in Onyx and before some recent change the `subStep` was not `null` (unless I'm misremembering that this was working fine recently).

As for the second half of this problem... to be honest, I think it should maybe be handled in another PR / issue. If you want enough time for the requests to all finish then we wouldn't have the issue. So it seems unrelated to the first problem and more related to the fact that we are making an extra http request.

aldo-expensify commented:
>Ok I read the first half of what you've wrote and I think I agree that it would be best to only rely on the achData Onyx to tell us which subStep we should show.

👍 

>We need to know the currentStep since bank accounts are saved in the server with information about which step the user needs to finish completing. If you check out the API layer this part will make more sense. As for the subStep I don't think the API cares about it (but not sure) and it's mostly a way to tell the front end to show the plaid or manual flows.

Ok, makes sense, I haven't looked into that code yet.

>As for the second half of this problem... to be honest, I think it should maybe be handled in another PR / issue. If you wait enough time for the requests to all finish then we wouldn't have the issue. So it seems unrelated to the first problem and more related to the fact that we are making an extra http request.

The double fetch was already there, but it didn't cause any visible problem because of the local state keeping a copy of the `subStep`, but with my change, it causes visible problems. I was fixing it here because my PR is causing this other problem to appear, but if you prefer, we can split it as you say.

marcaaron commented:
>The double fetch was already there, but it didn't cause any visible problem because of the local state keeping a copy of the subStep, but with my change, it causes visible problems. I was fixing it here because my PR is causing this other problem to appear, but if you prefer, we can split it as you say.

My preference will always be to fix one issue at a time mostly because if the one half of this has a regression then you don't have to roll back 100% of it. But also less complexity makes things move faster.

aldo-expensify committed:
Merge branch 'main' of github.com:Expensify/App into aldo_fix-wrong-substep-after-error

aldo-expensify committed:
Revert Onyx merge separate tick and double fetch

aldo-expensify committed:
Move Onyx.merge to actions

aldo-expensify committed:
Remove unwanted destructuring

aldo-expensify committed:
Remove empty line

aldo-expensify commented:
Moving the details about the double fetch here to clean up the PR's description. Also, this is not causing problems anymore, so it there it is low priority to do something about it now.


After the change I did to read the `subStep` always from what we had in Onyx, the following problem started to happen:

https://user-images.githubusercontent.com/87341702/132899997-34f650b9-7412-4327-9f9b-902562d3ee7d.mov

Basically, if you navigate to `/bank-account/new` and ""quickly"" click "Connect Manually", you get bounced back to the previous view.

This is happening because:

1. Fetch withdrawal account information when app starts ([here](https://github.com/Expensify/App/blob/894e68393134c87049dbd8d01029457ddbab0a35/src/libs/Navigation/AppNavigator/AuthScreens.js#L191))
2. Set `reimbursementAccount.loading` to `true` in Onyx ([here](https://github.com/Expensify/App/blob/6cdfd4b9003e74d4ea1806f79ce16485f778ae6e/src/libs/actions/BankAccounts.js#L615))
3. `ReimbursementAccountPage` component is created, in its constructor we fetch again the information for the withdrawal account ([here](https://github.com/Expensify/App/blob/6cdfd4b9003e74d4ea1806f79ce16485f778ae6e/src/pages/ReimbursementAccount/ReimbursementAccountPage.js#L92))
4. The first fetch comes back with the information, we set `reimbursementAccount.loading` to `false` ([here](https://github.com/Expensify/App/blob/6cdfd4b9003e74d4ea1806f79ce16485f778ae6e/src/libs/actions/BankAccounts.js#L652-L655)). The information fetched comes with `subStep = null`.
5. The user can see the options "Connect Manually" and "Log into your bank"
6. Clicking "Connect Manually" sets `reimbursementAccount.achData.subStep` in Onyx to `'manual'` ([here](https://github.com/Expensify/App/blob/251ee9719c37a3bd0259c89a31914bec0c964295/src/pages/ReimbursementAccount/BankAccountStep.js#L233), [here](https://github.com/Expensify/App/blob/251ee9719c37a3bd0259c89a31914bec0c964295/src/pages/ReimbursementAccount/BankAccountStep.js#L82-L90)), we show the form to input "Routing Number" and "Account Number"
7. The second fetch comes back, again with the `subStep = null`, overriding the value that the user just set when clicked "Connect Manually". This takes the user back to the first screen.

Ways to solve the second problem:
- If we are doing multiple fetches, increment some number somewhere that would invalidate the previous one if a new one is ongoing.
- Avoid doing a second fetch

I took the approach of avoiding to do a second fetch.

aldo-expensify commented:
Update:

- Removed the code related to avoiding double fetch of withdrawal account info
- Moved `Onyx.merge` code from `BankAccountStep.js` to action `BankAccounts.js`
- Fixed code styling

aldo-expensify committed:
Merge branch 'main' - Resolve conflicts

aldo-expensify committed:
Merge branch 'main' - resolve conflicts

aldo-expensify commented:
Update: update with main, resolve conflicts

aldo-expensify commented:
@marcaaron and @marcochavezf , just a reminder that this is ready for review!

marcaaron commented:
Hey @aldo-expensify just wanted to confirm if this is an N6 blocker?

aldo-expensify commented:
> Hey @aldo-expensify just wanted to confirm if this is an N6 blocker?

Sorry, I'm not sure about the criteria for something to be a N6 blocker, the issue at least is not labeled as such.

This is a bug in the VBA flow, but it doesn't block the user from completing the flow, it just bounces back once. After the user clicks a second time the button to add the bank account manually, it stops happening.

marcaaron commented on src/libs/actions/BankAccounts.js:
```diff
@@ -652,18 +661,8 @@ function setupWithdrawalAccount(data) {
-                .then(() => Promise.resolve(response));
-        })
         .then((response) => {
+            Onyx.merge(ONYXKEYS.REIMBURSEMENT_ACCOUNT, {loading: false, achData: {...newACHData}});
```
I'm admittedly a little nervous about this change. But only because I don't fully understand why the code we are removing was necessary in the first place.

The original comment says:

>Without this block, we can call merge again with the achData before this merge finishes, resulting in the original achData overwriting the data we're trying to set here.

But I'm not sure which additional call to `merge()` it's referring to. In addition, any merges made in this execution context should get batched together and resolve as one single "merge" on the Onyx side.

Putting that aside for now... is this change necessary in order to get the new `subStep` logic to work properly?

marcaaron replied:
Hmm did we talk about this already maybe? Sorry I don't think I really got it the first time 😄

aldo-expensify replied:
I think this is the change that fixes the whole problem.

Initially my solution was different:
- Avoid double fetch of withdrawal account information (removed from PR)
- Read `subStep` directly from Onyx instead of copying it to local state.

After more debugging, I saw that the commented code here was causing the `BankAccountStep` component to be rendered before the whole information was available. Rendering the `BankAccountStep` before the right time, caused it to get an incorrect subStep copied in its local state.

I think that If you remove this piece of code that the PR is intending to remove in `BankAccounts.js`, it is not necessary any more to read the `subStep` directly from Onyx, but I left this change there anyway because I can help preventing future bugs.

I can do more testing around this if you think it would be good.

aldo-expensify replied:
I tried to replicate the problem @Jag96 's code was fixing and it doesn't seem to be happening anymore. I'm thinking that it probably got fixed by some fix in Onyx. Another possibility is that it is not happening anymore because we introduced this "draft save" mechanism and I think we give priority to reading from there.

Jag96 replied:
> But I'm not sure which additional call to merge() it's referring to.

I should've specified, I don't see it now and I can't reproduce so not sure which other merge was getting batched. Maybe it was [this one](https://github.com/Expensify/App/pull/4477/files#diff-448c3537222b5e687caa8f06585ace6feae03d751f243e1590383fb6a33693e8L475).

> In addition, any merges made in this execution context should get batched together and resolve as one single "merge" on the Onyx side.

Merge 1 set the new bankAccountNumber, Merge 2 didn't set any bankAccountNumber, and when it resolved into one merge it was using the bankAccountNumber from Merge 2 which was the original one.

> I'm thinking that it probably got fixed by some fix in Onyx.

We should be able to test this pretty easily by calling Onyx.merge twice and seeing if the value set in the first merge is overwritten by the original value if the 2nd merge doesn't set that value.

marcaaron replied:
Ok sounds like this change should be fine. I tested a bunch and don't really see anything not working differently from `main`.

marcaaron reviewed (commented):
These changes look good to me, but I'm not 100% sure about the change in `BankAccounts.js` gonna do some testing and tag in @Jag96 to take a look at my comment.

marcaaron reviewed (commented):
Passes the initial tests for me, but I ran into an issue when performing the following steps:

1. Add manual account information (as specified in existing test steps)
2. Verified error message appeared
3. Navigated back to previous step
4. Selected the Plaid menu item
5. Signed into chase test bank
6. Error still shows..

![2021-09-23_09-59-14](https://user-images.githubusercontent.com/32969087/134575653-8fae1571-d203-46ab-b966-44409afb8af0.png)

7. Selected the savings account option and advanced to the company step.
8. Pressed to go back to previous step
9. Manual step shown (I'm not sure if that is expected or we should return to the Plaid screen?)

![2021-09-23_10-03-04](https://user-images.githubusercontent.com/32969087/134575927-a3f22fe1-9d0a-4af0-92e3-28deab0dcd29.png)

10. Enter a valid routing and account number (`acct: 011401533` `rtg: 1111222233331111`)
11. Advance to company step
12. Go back to previous step
13. Enter `11` for account number
14. We are able to advance to the company step now despite entering an invalid account number

Not sure if these are blockers or if they are happening on `main` as well. Going to test there next to be sure we don't have any regressions.

marcaaron commented:
Ok I can repro all that stuff on `main` so I don't think it has to do with anything here 😄

aldo-expensify commented:
> Ok I can repro all that stuff on `main` so I don't think it has to do with anything here 😄

😌 , do you want me to have a look at those other problems you found?

marcaaron commented:
>do you want me to have a look at those other problems you found

I would do something like:

1. Determine if they are blockers or polish for N6
2. See if we need to look at them urgently or not
3. See who is available to help or fix them ourselves

marcaaron commented:
Also can we add test steps to make sure this error is showing?

![2021-09-23_10-32-14](https://user-images.githubusercontent.com/32969087/134579700-17759139-15ae-4d45-94ac-ea7fcd819fab.png)

aldo-expensify commented:
> Also can we add test steps to make sure this error is showing?

Sure, to get there with an unverified account you have to do the OldDot -> NewDot flow, right?

marcaaron commented:
Yep! Left a comment here about this -> https://github.com/Expensify/Web-Secure/pull/2042#issuecomment-926155516

aldo-expensify commented:
> Yep! Left a comment here about this -> [Expensify/Web-Secure#2042 (comment)](https://github.com/Expensify/Web-Secure/pull/2042#issuecomment-926155516)

Cool, I didn't know exactly how to get there, those steps help :)

marcaaron commented:
Going to CP this but let's make sure to update steps and test the unvalidated flow on staging

aldo-expensify commented:
> Going to CP this but let's make sure to update steps and test the unvalidated flow on staging

yes yes, sorry for the slowness on that.. I'm still struggling to get from OldDot to NewDot


Following @robertjchen 's PR steps, after "Select Expensify Card option ONLY in WelcomeBusiness inbox task", I see this, what then 🤷 

![Screen Shot 2021-09-23 at 2 31 12 PM](https://user-images.githubusercontent.com/87341702/134586911-d2f23da8-7ef2-4d6c-8e61-251465577d30.png)

I tried a few options, but I haven't been able to see something that takes me to new Dot. If I just try to load NewDot using localhost:8080, I cannot log in because the account is not verified.

aldo-expensify commented:
Just saw that if I open it with a small screen (mobile), I see a button that takes me to NewDot :)

aldo-expensify commented:
Update: Added the "unverified account error" to the test / QA.

OSBotify commented:
🚀 [Cherry-picked](https://github.com/Expensify/App/actions/runs/1267584990) to staging by @marcaaron in version: 1.1.1-8 🚀

 platform | result 
 ---|--- 
🤖 android 🤖|success ✅ 
🖥 desktop 🖥|success ✅
🍎 iOS 🍎|success ✅ 
🕸 web 🕸|success ✅

@Expensify/applauseleads please QA this PR and check it off on the [deploy checklist](https://github.com/Expensify/App/issues?q=is%3Aopen+is%3Aissue+label%3AStagingDeployCash) if it passes.

kadiealexander commented:
>Update: Only found this mention https://expensify.slack.com/archives/C020EPP9B9A/p1631770824318400?thread_ts=1631562971.004600&cid=C020EPP9B9A where @kadiealexander was asking for assistance on figuring out if they were n6-hold or not.

It was decided that this is a blocker for n6, so a hold won't apply here.

kevinksullivan commented:
I was able to test this successfully! Can we check it off @applausebot  @isagoico ?

![image](https://user-images.githubusercontent.com/22508290/134596519-6614283b-1911-4813-ba5b-2923d2c78f8c.png)

OSBotify commented:
🚀 [Deployed](https://github.com/Expensify/App/actions/runs/1267880092) to production by @Jag96 in version: 1.1.1-8 🚀

 platform | result 
 ---|--- 
🤖 android 🤖|success ✅ 
🖥 desktop 🖥|success ✅
🍎 iOS 🍎|success ✅ 
🕸 web 🕸|success ✅

OSBotify commented:
🚀 [Deployed](https://github.com/Expensify/App/actions/runs/1267883267) to staging by @marcaaron in version: 1.1.1-9 🚀

 platform | result 
 ---|--- 
🤖 android 🤖|success ✅ 
🖥 desktop 🖥|success ✅
🍎 iOS 🍎|success ✅ 
🕸 web 🕸|success ✅

OSBotify commented:
🚀 [Deployed](https://github.com/Expensify/App/actions/runs/1283974506) to production by @timszot in version: 1.1.2-7 🚀

 platform | result 
 ---|--- 
🤖 android 🤖|success ✅ 
🖥 desktop 🖥|success ✅
🍎 iOS 🍎|failure ❌ 
🕸 web 🕸|success ✅
